app/uk/gov/hmrc/tariffclassificationfrontend/views/partials/keywords_details.scala.html

```diff
@@ -0,0 +1,90 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *@
+
+@import helper._
+@import uk.gov.hmrc.tariffclassificationfrontend.config.AppConfig
+@import uk.gov.hmrc.tariffclassificationfrontend.forms.KeywordFormData
+@import uk.gov.hmrc.tariffclassificationfrontend.models.request.AuthenticatedRequest
+@import uk.gov.hmrc.tariffclassificationfrontend.models.{Case, CaseStatusChange, Event, EventType}
+@import uk.gov.hmrc.tariffclassificationfrontend.utils.Dates
+@import uk.gov.hmrc.tariffclassificationfrontend.views.html.forms.components.input_text
+@import uk.gov.hmrc.tariffclassificationfrontend.views.html.partials.{progressive_disclosure, tab_heading}
+
+@import uk.gov.hmrc.tariffclassificationfrontend.views.html.components.input_autoComplete
+@(c: Case, autoCompleteOptions: Seq[String], keywordForm : Form[KeywordFormData])(implicit request: AuthenticatedRequest[_], messages: Messages, appConfig: AppConfig)
+
+    @tab_heading("keywords", messages("case.menu.keywords"))
+
+    <div class="grid-row">
+
+        <div class="column-full">
+        <div>
+            <p>
+            Cases must have 3 or more keywords.
+            </p>
+        </div>
+
+        <div>
+            <table id="keywords-table">
+                <tbody>
+                    @for((keyword: String, index: Int) <- c.keywords.toSeq.sorted.zipWithIndex) {
+                        <tr id="keywords-row-@{index}">
+                            <td id="keywords-row-@{index}-keyword" class="w-50">@keyword</td>
+
+                            <td id="keywords-row-@{index}-message" class="w-40">
+                                @if(!autoCompleteOptions.contains(keyword)) {
+                                    This is not from the main list of keywords
+                                }
+                            </td>
+
+                            <td id="keywords-row-@{index}-remove" >
+                                <a href="@uk.gov.hmrc.tariffclassificationfrontend.controllers.routes.CaseController.removeKeyword(c.reference, keyword)">Remove</a>
+                            </td>
+                        </tr>
+                    }
+                </tbody>
+
+            </table>
+
+        </div>
+
+
+        @form(action = uk.gov.hmrc.tariffclassificationfrontend.controllers.routes.CaseController.addKeyword(c.reference), 'class -> "form") {
+
+            @CSRF.formField
+
+            @input_autoComplete(
+                field = keywordForm("keyword"),
+                label = "Add a keyword to this case",
+                autoCompleteOptions = autoCompleteOptions
+            )
+
+            <div class="form-group">
+                <button id="add-keyword-submit" class="button" type="submit">Add keyword</button>
```

Comment:
Up for debate but I think we should have an ID naming convention e.g.
`module_name-part_of_page-element`

Reply:
Agreed. Done